db/friend.go

```diff
@@ -1,5 +1,84 @@
 package db
 
+import (
+	"fmt"
+	"log"
+)
+
 type Friend struct {
-	MAC, IP string
+	DisplayName, MAC, IP string
+}
+
+// Checks if the two users are friends given friend's display name
+func AreFriendsName(username string, friendDisplayName string) bool {
+	query := fmt.Sprintf("SELECT * FROM %s WHERE username=\"%s\" AND friend_display_name=\"%s\"", friendsTableName, username, friendDisplayName)
+	friends := ExecuteFriendsQuery(query)
+	return len(friends) > 0
+}
+
+// Checks if the two users are friends given friend's MAC Address
+func AreFriendsMac(username string, friendMacAddress string) bool {
+	query := fmt.Sprintf("SELECT * FROM %s WHERE username=\"%s\" AND friend_mac_address=\"%s\"", friendsTableName, username, friendMacAddress)
+	friends := ExecuteFriendsQuery(query)
+	return len(friends) > 0
+}
+
+// Get friend from database
+func GetFriendByDisplayName(friendDisplayName string) *Friend {
+	query := fmt.Sprintf("SELECT friend_display_name, friend_mac_address, friend_ip_address FROM %s WHERE friend_display_name= \"%s\"", friendsTableName, friendDisplayName)
+	friends := ExecuteFriendsQuery(query)
+	if len(friends) == 0 {
+		return nil
+	}
+	return &friends[0]
+}
+
+// Add friend to the user's friend's table
+func AddFriend(username string, displayName string, macAddress string, ipAddress string) bool {
+	log.Println("Inserting data into friends...")
+	insertCommand := fmt.Sprintf("INSERT INTO %s VALUES (\"%s\", \"%s\", \"%s\")", friendsTableName, username, displayName, macAddress, ipAddress)
+	return ExecuteChangeCommand(insertCommand, "Failed to add friend")
+}
+
+// Delete a friend by their display name
+func DeleteFriendByDisplayName(friendName string) bool {
+	deleteCommand := fmt.Sprintf("DELETE FROM %s WHERE friend_display_name= \"%s\"", friendsTableName, friendName)
+	return ExecuteChangeCommand(deleteCommand, "Failed to delete friend")
+}
+
+// Delete a friend by their MAC Address
+func DeleteFriendByMac(macAddress string) bool {
+	deleteCommand := fmt.Sprintf("DELETE FROM %s WHERE friend_mac_address= \"%s\"", friendsTableName, macAddress)
+	return ExecuteChangeCommand(deleteCommand, "Failed to delete friend")
+}
+
+// Get all friends
+func QueryFriends() []Friend {
```

Comment:
Shouldn't this function be: get all of a user's friends? Since technically multiple users can user this chat app.

Reply:
Yes it should  be.